pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+    ######################################################################
+
+    def _fake_enumerateclasses(self, namespace, **params):
+        """
+        Implements a mock equivalent of
+        :meth:`~pywbem.WBEMConnection.EnumerateClasses`.
+
+        Enumerate classes from class dictionary. If classname parameter
+        exists, use it as the starting point fin the hiearchy to get subclasses
+
+        """
+        self._validate_classes_namespace(namespace)
+
+        cn = params.get('classname', None)
+
+        cns = self._get_subclass_names(
+            cn, namespace, deep_inheritance=params['DeepInheritance'])
+        try:
+            del params['classname']
+        except KeyError:
+            pass
+        classes = []
+        for cn in cns:
+            classes.append(self._get_class(cn, namespace, **params))
+
+        return self._make_tuple(classes)
+
+    def _fake_enumerateclassnames(self, namespace, **params):
+        """
+        Implements a mock equivalent of
+        :meth:`~pywbem.WBEMConnection.EnumerateClassNames`.
+
+        Returns:
+            returns classnames.
+        Exceptions:
+            invalid namespace,
+            Classname not found
+        """
+        self._validate_classes_namespace(namespace)
+
+        cns = self._get_subclass_names(
+            params.get('classname', None),
+            namespace,
+            deep_inheritance=params['DeepInheritance'])
+
+        rtn_cns = [CIMClassName(cn, namespace=namespace, host=self.host)
+                   for cn in cns]
+        return self._make_tuple(rtn_cns)
+
+    def _fake_getclass(self, namespace, **params):
+        """
+        Implements a mock equivalent of
+        :meth:`~pywbem.WBEMConnection.GetClass
+
+        Retrieve a CIM class from the local repository.
+
+        For a description of the parameters, see
+        :meth:`pywbem.WBEMConnection.GetClass`.
+        """
+        self._validate_classes_namespace(namespace)
+        cname = params['ClassName'].classname
+
+        cc = self._get_class(cname, namespace, **params)
+        # TODO need to get away from having to put list around single return
+        return self._make_tuple([cc])
+
+    def _fake_createclass(self, namespace, **params):
+        """
+        Implements a mock server reactor for
+
+        :meth:`~pywbem.WBEMConnection.CreateClass`
+
+        Creates a new class in the repository.  Nothing is returned.
+        Emulates WBEMConnection.CreateClass(...))
+
+        if the class repository for this namespace does not
+        exist, this method creates it.
+
+        Returns:
+            None
+        Exceptions:
+            CIMError
+        """
+        new_class = params['NewClass']
+        if namespace not in self.classes:
+            self.classes[namespace] = NocaseDict({})
+
+        if not isinstance(new_class, CIMClass):
+            CIMError(CIM_ERR_INVALID_PARAMETER, 'NewClass not valid class\n%s'
+                     % new_class)
```

Comment:
Since this is a check about the type, I'd like to see the type shown as in `type(obj)`, and not `str(obj)`.

Reply:
Done